middlewares/countriesMiddlewares.js

```diff
@@ -0,0 +1,36 @@
+// const countriesRepositories = require('../repositories/countriesRepositories');
+const countriesRepositories = require('../repositories/countriesRepositories');
+const countriesUtilities = require('../utilities/countriesUtilities');
+
+exports.countryExists = (req, res, next) => {
+    const availableCountries = countriesRepositories.getCountries(req.params.name);
```

Comment:
Ar fi indicat ca in middlewares sa nu accesam baza de date a resurselor din proeict, nu intra in scopul unui middleware sa faca asta neaparat. Daca un country exista sau nu ii revine controller-ului sau ce e sub controller sa verifice si sa reactioneze corespunzator.